bundles/binding/org.openhab.binding.rwesmarthome/src/main/java/org/openhab/binding/rwesmarthome/internal/RWESmarthomeGenericBindingProvider.java

```diff
@@ -0,0 +1,217 @@
+				}
+				config.setDeviceId(value);
+				
+			// param
+			} else if("param".equalsIgnoreCase(key)){
+				if(!value.matches("^(temperature|humidity|settemperature|variable|contact|switch|operationmodeauto|luminance|smokedetector|dimmer|dimmerinverted|rollershutter|rollershutterinverted|alarm)$")) {
+					throw new BindingConfigParseException("Invalid configuration: 'param' must be one of the following: temperature|humidity|settemperature|variable|contact|switch|operationmodeauto|luminance|smokedetector|dimmer|dimmerinverted|rollershutter|rollershutterinverted|alarm");
+				}
+				config.setDeviceParam(value);
+		
+			// unknown configuration key
+			} else {
+				logger.warn("Invalid configuration key '%s' - only 'id' and 'param' are allowed!", key, value);
+				throw new BindingConfigParseException("Invalid configuration key '" + key + "'");
+			}
+		}
+		
+		if(config.getDeviceId() == null)
+			throw new BindingConfigParseException("Invalid configuration: id is missing!");
+		if(config.getDeviceParam() == null)
+			throw new BindingConfigParseException("Invalid configuration: param is missing!");
+		
+		logger.info("Adding item {} with {}", item.getName(), config.toString());
+		items.put(item.getName(), item);
+		
+		itemMapById.put(config.getDeviceId(), item);
+		itemMapByIdAndParam.put(config.getDeviceId(), config.getDeviceParam(), item);
+		
+		addBindingConfig(item, config);
+		this.context.setBindingChanged(true);
+	}
+	
+	
+	/**
+	 * This is a helper class holding binding specific configuration details
+	 * 
+	 * @author ollie-dev
+	 * @since 1.8.0
+	 */
+	public class RWESmarthomeBindingConfig implements BindingConfig {
+		private String deviceId = null;
+		private String deviceParam = null;
+		
+		/**
+		 * @return the device id
+		 */
+		public String getDeviceId() {
+			return deviceId;
+		}
+
+		/**
+		 * Sets the deviceId
+		 * 
+		 * @param deviceId
+		 */
+		public void setDeviceId(String deviceId) {
+			this.deviceId = deviceId;
+		}		
+		
+		/**
+		 * @return device parameter
+		 */
+		public String getDeviceParam() {
+			return deviceParam;
+		}
+
+		/**
+		 * @param deviceParam
+		 */
+		public void setDeviceParam(String deviceParam) {
+			this.deviceParam = deviceParam;
+		}
+
+		@Override
+		public String toString() {
+			String configString = "RWESmarthomeBindingConfig [deviceId=" + deviceId + ", param=" + deviceParam + "]";
+			return configString;
+		}
+
+	}
+
+
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider#getItem(java.lang.String)
+	 */
+	@Override
+	public Item getItem(String itemName) {
+		return items.get(itemName);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider#getItemById(java.lang.String)
+	 */
+	@Override
+	public Item getItemById(String id) {
+		return (Item) itemMapById.get(id);
+	}
```

Comment:
the class should never return an Item but rather the specific information needed from the Item (e.g. type)